src/store/fileUploadStore.js

```diff
@@ -66,6 +67,16 @@ const getters = {
 			.filter(([_index, uploadedFile]) => uploadedFile.status === 'initialised')
 	},
 
+	getFailedUploads: (state, getters) => (uploadId) => {
+		return getters.getUploadsArray(uploadId)
+			.filter(([_index, uploadedFile]) => uploadedFile.status === 'failedUpload')
```

Comment:
As I see, you are not using index in any of `getUploadsArray` calls so maybe it's better to change the output format 
```
 return Object.values(state.uploads[uploadId].files)
```

Reply:
`getUploadsArray` is acting as a middleware to check for existing of files. `index` is used in dependent functions:

```js
for (const [index, uploadedFile] of getters.getInitialisedUploads(uploadId)) {
context.getters.getFailedUploads(uploadId).forEach(([index, file]) => {
const uploads = getters.getUploadingFiles(uploadId)
const shares = getters.getShareableFiles(uploadId)
```